src/axom/primal/operators/detail/evaluate_integral_impl.hpp

```diff
@@ -57,13 +59,69 @@ inline double evaluate_scalar_line_integral_component(const primal::BezierCurve<
   return full_quadrature;
 }
 
+/*!
+ * \brief Evaluate a scalar field line integral on a single NURBS curve.
+ *
+ * Decompose the NURBS curve into Bezier segments, then sum the integral on each
+ *
+ * \param [in] n The NURBS curve object
+ * \param [in] integrand the lambda function representing the integrand. 
+ * Must accept a 2D point as input and return a double
+ * \param [in] npts The number of quadrature points in the rule
+ * \return the value of the integral
+ */
+template <class Lambda, typename T, int NDIMS>
+inline double evaluate_scalar_line_integral_component(const primal::NURBSCurve<T, NDIMS>& n,
+                                                      Lambda&& scalar_integrand,
+                                                      const int npts)
+{
+  const auto beziers = n.extractBezier();
+  double total_integral = 0.0;
+  for(int i = 0; i < beziers.size(); ++i)
+  {
+    total_integral +=
+      detail::evaluate_scalar_line_integral_component(beziers[i], scalar_integrand, npts);
```

Comment:
It may be good to add `std::forward<Lambda>(scalar_integrand)` here and in other places like this.